### PR TITLE
Allow for maker.args to be a string, skipping quotation/escaping

### PR DESCRIPTION
**Pull request by blueyed**
Fixes https://github.com/neomake/neomake/issues/837.
Closes https://github.com/neomake/neomake/pull/841.

TODO
 - [ ] probably does not work as expected on Windows?!  Could somebody test it?

### CODE REVIEW

File: autoload/neomake.vim [hunk 1]
```diff
@@ -121,12 +121,16 @@ function! s:MakeJob(make_id, maker) abort
...
-    if append_file
-        call add(args, '%:p')
-    endif
+    let args_is_list = type(args) == type([])
 
-    call neomake#utils#ExpandArgs(args)
+    if a:maker.file_mode && get(a:maker, 'append_file', 1)
+        if args_is_list
+            call neomake#utils#ExpandArgs(args)
+            call add(args, expand('%:p'))
```
Comment (bounceme) [inline autoload/neomake.vim]: what about 
```
       elseif args =~# '%:p'
            let args = substitute(args,'%:p','\=expand(submatch(0))','g')
```

Comment (bounceme) [inline autoload/neomake.vim]: in addition

**Comment by bounceme** [comment]
sorry can't test now. I'll be back at a computer in a few hours

File: autoload/neomake.vim (hunk 1 above)
Comment (bounceme) [inline autoload/neomake.vim]: oh right, forgot it can be a funcref

Comment (blueyed) [inline autoload/neomake.vim]: Yes, and also with `append_file`, it will get appended still, of course.

**Comment by bounceme** [comment]
```
function s:EF()
  return join([expand('%:p'),'--quick','--batch',
        \"--eval=\"(progn(package-initialize)(require 'js2-mode)(js2-mode)(js2-reparse t)(js2-display-error-list)(princ(with-current-buffer \\\"*js-lint*\\\"(buffer-substring-no-properties(point-min)(point-max))))(terpri))\"" ,
        \ '2>/dev/null'])
endfunction
let g:neomake_javascript_emacs_maker = {
      \ 'args': function('s:EF'),
      \ 'append_file': 0,
      \ 'errorformat': '%f line %l: %m',
      \ 'mapexpr': 'expand("%:p") . " " . v:val',
      \ }
```
working fine!

**Comment by bounceme** [comment]
and vim-async magically works!

**Comment by blueyed** [comment]
Great.

I will merge this then (soonish).

btw: it looks like you have to do a lot to work around not getting `stderr`, and after all we should probably not consider `stderr` for the `errorformat` processing, but rather use it as an indication that the maker failed.  I am creating an issue for that.

**Comment by bounceme** [comment]
thanks!

**Pull request merged**